CodeModel/CodeParsers/CSharp/ClassCohesionAnalyzer/CohesiveParts.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace CodeModel.CodeParsers.CSharp.ClassCohesionAnalyzer
+{
+    public class CohesiveParts
+    {
+        public List<HashSet<Access>> Parts { get; }
```

Comment:
Refactoring proposition: create a wrapper class around `HashSet<Access>`. Many methods have this collection as a parameter. Also by doing it I would avoid using `List<HashSet<Access>>`, and have a more meaningful name.
I am not sure about the class name. I was thinking about `Part` since it represents a part of a class, could be whole class, could be few method-field interactions. Maybe `ClassPart` since `Part` is too generic?

Reply:
This could probably host the GetAccessesThatCanBeCut and similar methods?

Reply:
Yes, most of the methods that require or work with `HashSet<Access>`.
One more thing, when I do refactor this, class `ClassInteractions` will only have a responsibility to create the initial `HashSet<Access>`, and I'm thinking of turning it into a static factory method. Thoughts?

Reply:
The static factory method can be part of the ClassPart class :)